crates/pagecache/src/ds/pagetable.rs

```diff
@@ -214,13 +211,13 @@ fn traverse<'g, T: 'static + Send>(
     let l1 = unsafe { &head.deref().children };
 
     debug_delay();
-    let mut l2_ptr = l1[usize::try_from(l1k).unwrap()].load(Acquire, guard);
+    let mut l2_ptr = l1[l1k as usize].load(Acquire, guard);
```

Comment:
I feel a little uncertain about using `as` instead of `try_from` because I like the guarantee that the `try_from` will panic when `unwrap`ped if the conversion fails. Maybe we can avoid any casting and return a `(usize, usize)` from `split_fanout` and also perform an assertion in there, to prevent the concerns from spreading all over the module?

Reply:
I thought you used it for a reason, but then I have checked the implementation that is:

```
// Infallible conversions are semantically equivalent to fallible conversions
// with an uninhabited error type.
impl<T, U> TryFrom<U> for T where U: Into<T> {
    type Error = Infallible;

    fn try_from(value: U) -> Result<Self, Self::Error> {
        Ok(U::into(value))
    }
}
```

and then I got lost a little because this long form of conversion literally do nothing. ?

Reply:
done

Reply:
For casting between u64 and usize, it will fail on 32-bit platforms when the u64 is bigger than the max u32

Reply:
oh, there are so much of them was there

Reply:
got it, they make it through the macros defs and cond compilation